mautic/mautic
# Marketplace version fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | https://github.com/mautic/marketplace-allowlist/pull/3
| Issue(s) addressed                     | 



#### Description:

With the release of the Custom Objects plugin we found that the Marketplace could handle some things better. So this PR fixes:
1. It takes the latest stable version and fallbacks to a feature branch if stable release does not exists.
2. Lets developers set a human readable name. Composer.json allows only for vendor/repository as the name.
3. The Issue Tracker button was not working. When in there I also added the wiki/documentation button.
4. I reordered the buttons a little. The install and the Close button is visible all the time, the Issue Tracker and Documentation are hidden in the context menu. It used to show the Install and the Issue Tracker button and Close being in the context menu.
5. Adds new "Clear Cache" button to the list of packages as I got an issue during development that an empty array got cached and I had no option to clear the cache other than waiting for an hour so the cache would invalidate. I think it could be useful for users too if they wait for new version of a package for example. And we should cache more stuff in the future.
6. Related to the point 5., I noticed that if there are no packages available then the message does not make sense. I fixed that like so:
![Screen Shot 2022-05-31 at 10 39 06](https://user-images.githubusercontent.com/1235442/171204256-d96d235b-33e3-450a-abd2-e71ab0732262.png)

Detail before:
![Screen Shot 2022-06-01 at 13 27 19](https://user-images.githubusercontent.com/1235442/171394201-00b1a6ab-d798-4864-ab40-dd6e7e77e461.png)
Detail after:
![Screen Shot 2022-06-01 at 13 28 43](https://user-images.githubusercontent.com/1235442/171394240-dd146564-b968-401a-8273-1a39922e2c47.png)
Notice: the buttons are in different order, the issue tracker button works, the versions do not have a link unless they are stable as otherwise the link goes to a 404 page. The plugin name is more human readable. The version release date contains time.


List before:
![Screen Shot 2022-06-01 at 13 27 35](https://user-images.githubusercontent.com/1235442/171394274-c42daa87-36e1-4551-97b0-512a972c4b01.png)
List after:
![Screen Shot 2022-06-01 at 13 28 30](https://user-images.githubusercontent.com/1235442/171394308-2037765e-664c-4043-86df-099ac7641c39.png)
Notice: The Custom Objects plugin name is human readable, the Clear Cache button appeared.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
8. Browse the Marketplace and notice the fixes in the description.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11197"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
